### PR TITLE
Adding authorization subsystem documentation

### PR DESCRIPTION
**Pull request by dimastopel**
This PR contains the docs changes for the AuthZ changes discussed here: [#14674](https://github.com/docker/docker/issues/14674).

Your comments are very welcome. 

Signed-off-by: Dima Stopel dima@twistlock.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks @dimastopel !

**Comment by tiborvass** [comment]
FYI there's also docker/swarm#1366 dealing with this, we should coordinate.

File: docs/extend/authorization.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+* The authorization sub-system enables approving or denying requests to the Docker daemon based on the current user and command context, where the authentication context contains all user details and the authentication method, and the command context contains all the relevant request data.
+
+* The authorization sub-system enables approving or denying the response from the Docker daemon based on the current user and the command context.
+
+## Basic architecture
+
+Authorization is integrated into Docker daemon by enabling external authorization plug-ins registration as part of the daemon stratup. Each plug-in will receive user's information (retrieved using the authentication sub-system) and HTTP request information and decide whether to allow or deny the request. Plugins can be chained together and only in case where all plug-ins allow accessing the resource, the is access granted.
```
Comment (SvenDowideit) [inline docs/extend/authorization.md]: `s/stratup/startup/` :)

**Comment by NathanMcCauley** [comment]
Design LGTM

**Comment by LK4D4** [comment]
Never saw word `authz` before :)

**Comment by calavera** [comment]
The changes in this PR are included as part of the final implementation in https://github.com/docker/docker/pull/15365. We don't need this separated PR anymore, so I'm closing it.

Thanks! :tada:

**Comment by dimastopel** [comment]
Sure

**Pull request closed without merging**